core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java

```diff
@@ -62,7 +66,16 @@ private enum State { CREATED, INITIALIZED, STOPPED }
     private final Executor executor;
 
     private final Lock catalogsUpdateLock = new ReentrantLock();
-    private final ConcurrentMap<String, CatalogConnector> catalogs = new ConcurrentHashMap<>();
+
+    /**
+     * Active catalogs that have been created and not dropped.
+     */
+    private final ConcurrentMap<String, CatalogConnector> activeCatalogs = new ConcurrentHashMap<>();
+
+    /**
+     * All catalogs including those that have been dropped.
+     */
+    private final ConcurrentMap<CatalogHandle, CatalogConnector> allCatalogs = new ConcurrentHashMap<>();
```

Comment:
Why do you need to distinguish `active` from `all` at this point? `active` never has any remove operations, so `active` and `all` are the same today.